--- qt-everywhere-opensource-src-4.7.1_orig/configure	2011-07-08 17:09:20.616358991 +0700
+++ qt-everywhere-opensource-src-4.7.1/configure	2011-07-08 17:07:23.108358992 +0700
@@ -6017,8 +6017,9 @@
                QT_CFLAGS_DIRECTFB=`$PKG_CONFIG --cflags directfb 2>/dev/null`
                QT_LIBS_DIRECTFB=`$PKG_CONFIG --libs directfb 2>/dev/null`
            elif directfb-config --version >/dev/null 2>&1; then
-               QT_CFLAGS_DIRECTFB=`directfb-config --cflags 2>/dev/null`
-               QT_LIBS_DIRECTFB=`directfb-config --libs 2>/dev/null`
+#               QT_CFLAGS_DIRECTFB=`directfb-config --cflags 2>/dev/null`
+#               QT_LIBS_DIRECTFB=`directfb-config --libs 2>/dev/null`
+                :;
            fi
 
            # QMake variables set here override those in the mkspec. Therefore we only set the variables here if they are not zero.
