--- accel-pptp-0.8.5.orig/kernel/driver/pptp.c	2010-10-06 20:27:46.000000000 +0700
+++ accel-pptp-0.8.5/kernel/driver/pptp.c	2012-06-04 16:53:46.000000000 +0700
@@ -30,6 +30,7 @@
 #include <linux/netfilter.h>
 #include <linux/netfilter_ipv4.h>
 #include <linux/version.h>
+#include <linux/vmalloc.h>
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
 #include <asm/bitops.h>
@@ -1145,14 +1146,8 @@
 	int err=0;
 	printk(KERN_INFO "PPTP driver version " PPTP_DRIVER_VERSION "\n");
 
-#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,0)
-	callid_sock = __vmalloc((MAX_CALLID + 1) * sizeof(void *),
-	                        GFP_KERNEL | __GFP_ZERO, PAGE_KERNEL);
-#else
-	callid_sock = __vmalloc((MAX_CALLID + 1) * sizeof(void *),
-	                        GFP_KERNEL, PAGE_KERNEL);
+	callid_sock = vmalloc((MAX_CALLID + 1) * sizeof(void *));
 	memset(callid_sock, 0, (MAX_CALLID + 1) * sizeof(void *));
-#endif
 	if (!callid_sock) {
 		printk(KERN_ERR "PPTP: cann't allocate memory\n");
 		return -ENOMEM;
