--- linux-sh4-2.6.32.57_stm24_V5.0_orig/drivers/media/dvb/frontends/Kconfig	2001-05-09 04:18:28.000000000 +0700
+++ linux-sh4-2.6.32.57_stm24_V5.0/drivers/media/dvb/frontends/Kconfig	2012-05-25 20:17:42.124186808 +0700
@@ -349,6 +349,13 @@
 	help
 	  A DVB-T/C tuner module. Say Y when you want to support this frontend.
 
+config DVB_CXD2820R
+	tristate "Sony CXD2820R"
+	depends on DVB_CORE && I2C
+	default m if DVB_FE_CUSTOMISE
+	help
+	  Say Y when you want to support this frontend.
+
 comment "DVB-C (cable) frontends"
 	depends on DVB_CORE
 
--- linux-sh4-2.6.32.57_stm24_V5.0_orig/drivers/media/dvb/frontends/Makefile	2001-05-09 04:18:28.000000000 +0700
+++ linux-sh4-2.6.32.57_stm24_V5.0/drivers/media/dvb/frontends/Makefile	2012-05-25 20:20:33.976177706 +0700
@@ -9,6 +9,7 @@
 stb0899-objs = stb0899_drv.o stb0899_algo.o
 stv0900-objs = stv0900_core.o stv0900_sw.o
 au8522-objs = au8522_dig.o au8522_decoder.o
+cxd2820r-objs = cxd2820r_core.o cxd2820r_c.o cxd2820r_t.o cxd2820r_t2.o
 
 obj-$(CONFIG_DVB_PLL) += dvb-pll.o
 obj-$(CONFIG_DVB_STV0299) += stv0299.o
@@ -77,3 +78,4 @@
 obj-$(CONFIG_DVB_STV6110x) += stv6110x.o
 obj-$(CONFIG_DVB_ISL6423) += isl6423.o
 obj-$(CONFIG_DVB_STV0367) += stv0367.o
+obj-$(CONFIG_DVB_CXD2820R) += cxd2820r.o
--- linux-sh4-2.6.32.57_stm24_V5.0_orig/include/linux/dvb/frontend.h	2001-05-09 04:19:05.000000000 +0700
+++ linux-sh4-2.6.32.57_stm24_V5.0/include/linux/dvb/frontend.h	2012-05-25 20:17:42.128186807 +0700
@@ -174,7 +174,10 @@
 	TRANSMISSION_MODE_2K,
 	TRANSMISSION_MODE_8K,
 	TRANSMISSION_MODE_AUTO,
-	TRANSMISSION_MODE_4K
+	TRANSMISSION_MODE_4K,
+	TRANSMISSION_MODE_1K,
+	TRANSMISSION_MODE_16K,
+	TRANSMISSION_MODE_32K,
 } fe_transmit_mode_t;
 
 typedef enum fe_bandwidth {
@@ -190,7 +193,10 @@
 	GUARD_INTERVAL_1_16,
 	GUARD_INTERVAL_1_8,
 	GUARD_INTERVAL_1_4,
-	GUARD_INTERVAL_AUTO
+	GUARD_INTERVAL_AUTO,
+	GUARD_INTERVAL_1_128,
+	GUARD_INTERVAL_19_128,
+	GUARD_INTERVAL_19_256,
 } fe_guard_interval_t;
 
 
@@ -321,7 +327,7 @@
 
 typedef enum fe_delivery_system {
 	SYS_UNDEFINED,
-	SYS_DVBC_ANNEX_AC,
+	SYS_DVBC_ANNEX_A,
 	SYS_DVBC_ANNEX_B,
 	SYS_DVBT,
 	SYS_DSS,
@@ -336,8 +342,11 @@
 	SYS_DMBTH,
 	SYS_CMMB,
 	SYS_DAB,
+	SYS_DVBT2,
 } fe_delivery_system_t;
 
+#define SYS_DVBC_ANNEX_AC       SYS_DVBC_ANNEX_A
+
 struct dtv_cmds_h {
 	char	*name;		/* A display name for debugging purposes */
 
