--- linux-sh4-2.6.32.57_stm24_V5.0_orig/include/linux/dvb/frontend.h	2001-05-09 04:19:05.000000000 +0700
+++ linux-sh4-2.6.32.57_stm24_V5.0/include/linux/dvb/frontend.h	2012-05-25 20:17:42.128186807 +0700
@@ -174,7 +174,10 @@
 	TRANSMISSION_MODE_2K,
 	TRANSMISSION_MODE_8K,
 	TRANSMISSION_MODE_AUTO,
-	TRANSMISSION_MODE_4K
+	TRANSMISSION_MODE_4K,
+	TRANSMISSION_MODE_1K,
+	TRANSMISSION_MODE_16K,
+	TRANSMISSION_MODE_32K,
 } fe_transmit_mode_t;
 
 typedef enum fe_bandwidth {
@@ -190,7 +193,10 @@
 	GUARD_INTERVAL_1_16,
 	GUARD_INTERVAL_1_8,
 	GUARD_INTERVAL_1_4,
-	GUARD_INTERVAL_AUTO
+	GUARD_INTERVAL_AUTO,
+	GUARD_INTERVAL_1_128,
+	GUARD_INTERVAL_19_128,
+	GUARD_INTERVAL_19_256,
 } fe_guard_interval_t;
 
 
@@ -321,7 +327,7 @@
 
 typedef enum fe_delivery_system {
 	SYS_UNDEFINED,
-	SYS_DVBC_ANNEX_AC,
+	SYS_DVBC_ANNEX_A,
 	SYS_DVBC_ANNEX_B,
 	SYS_DVBT,
 	SYS_DSS,
@@ -336,8 +342,11 @@
 	SYS_DMBTH,
 	SYS_CMMB,
 	SYS_DAB,
+	SYS_DVBT2,
 } fe_delivery_system_t;
 
+#define SYS_DVBC_ANNEX_AC       SYS_DVBC_ANNEX_A
+
 struct dtv_cmds_h {
 	char	*name;		/* A display name for debugging purposes */
 
