--- linux-sh4-2.6.32.57_stm24_V5.0_orig/drivers/mtd/nand/nand_ids.c	2012-11-21 20:12:03.730101005 +0700
+++ linux-sh4-2.6.32.57_stm24_V5.0/drivers/mtd/nand/nand_ids.c	2012-11-21 20:12:06.914101015 +0700
@@ -109,6 +109,8 @@
 	{"NAND 1GiB 3,3V 8-bit",	0xD3, 0, 1024, 0, LP_OPTIONS},
 	{"NAND 1GiB 1,8V 16-bit",	0xB3, 0, 1024, 0, LP_OPTIONS16},
 	{"NAND 1GiB 3,3V 16-bit",	0xC3, 0, 1024, 0, LP_OPTIONS16},
+//MT29F8G08ABABA
+	{"NAND 1GiB 3,3V 8-bit", 	0x38, 0, 1024, 0, LP_OPTIONS},
 
 	/* 16 Gigabit */
 	{"NAND 2GiB 1,8V 8-bit",	0xA5, 0, 2048, 0, LP_OPTIONS},
