diff --git a/xl2tpd.c b/xl2tpd.c
index 3fb6dd7..d863a49 100644
--- a/xl2tpd.c
+++ b/xl2tpd.c
@@ -421,6 +421,17 @@ int start_pppd (struct call *c, struct ppp_opts *opts)
        stropt[pos] = (char *) malloc (10);
        snprintf (stropt[pos], 10, "%d", fd2);
         pos++;
+       if (c->container->lns) {
+        stropt[pos++] = strdup ("pppol2tp_lns_mode");
+        stropt[pos++] = strdup ("pppol2tp_tunnel_id");
+        stropt[pos] = (char *) malloc (10);
+        snprintf (stropt[pos], 10, "%d", c->container->ourtid);
+            pos++;
+        stropt[pos++] = strdup ("pppol2tp_session_id");
+        stropt[pos] = (char *) malloc (10);
+        snprintf (stropt[pos], 10, "%d", c->ourcid);
+            pos++;
+       }
         stropt[pos] = NULL;
     }
     else
