--- hostapd-0.7.3/src/ap/hw_features.c.orig	2012-06-26 10:59:17.883477506 +0700
+++ hostapd-0.7.3/src/ap/hw_features.c	2012-06-26 10:59:26.106350535 +0700
@@ -422,6 +422,7 @@
 
 static void ieee80211n_check_scan(struct hostapd_iface *iface)
 {
+#if 0
 	struct wpa_scan_results *scan_res;
 	int oper40;
 
@@ -451,6 +452,7 @@
 		iface->conf->secondary_channel = 0;
 		iface->conf->ht_capab &= ~HT_CAP_INFO_SUPP_CHANNEL_WIDTH_SET;
 	}
+#endif
 
 	hostapd_setup_interface_complete(iface, 0);
 }
